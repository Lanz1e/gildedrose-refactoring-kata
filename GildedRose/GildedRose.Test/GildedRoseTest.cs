using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using GildedRose.Application.Services;
using GildedRose.Domain.Models.Base;
using GildedRose.Domain.Models.Items;
using NUnit.Framework;
using System.Collections.Generic;

namespace GildedRose.Test
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class GildedRoseTest
    {
        [Test]
        public void UpdateQuality()
        {
            IList<CompleteItem> testList = new List<CompleteItem> {
                new Default("+5 Dexterity Vest", -1, 0),
                new Default("+5 Dexterity Vest", -1, 1),
                new Default("+5 Dexterity Vest", -1, 49),
                new Default("+5 Dexterity Vest", -1, 51),
                new Default("+5 Dexterity Vest", 1, 0),
                new Default("+5 Dexterity Vest", 1, 1),
                new Default("+5 Dexterity Vest", 1, 49),
                new Default("+5 Dexterity Vest", 1, 51),
                new Default("+5 Dexterity Vest", 5, 0),
                new Default("+5 Dexterity Vest", 5, 1),
                new Default("+5 Dexterity Vest", 5, 49),
                new Default("+5 Dexterity Vest", 5, 51),
                new Default("+5 Dexterity Vest", 7, 0),
                new Default("+5 Dexterity Vest", 7, 1),
                new Default("+5 Dexterity Vest", 7, 49),
                new Default("+5 Dexterity Vest", 7, 51),
                new Default("+5 Dexterity Vest", 11, 0),
                new Default("+5 Dexterity Vest", 11, 1),
                new Default("+5 Dexterity Vest", 11, 49),
                new Default("+5 Dexterity Vest", 11, 51),
                new AgedBrie("Aged Brie",-1,0),
                new AgedBrie("Aged Brie",-1,1),
                new AgedBrie("Aged Brie",-1,49),
                new AgedBrie("Aged Brie",-1,51),
                new AgedBrie("Aged Brie",1,0),
                new AgedBrie("Aged Brie",1,1),
                new AgedBrie("Aged Brie",1,49),
                new AgedBrie("Aged Brie",1,51),
                new AgedBrie("Aged Brie",5,0),
                new AgedBrie("Aged Brie",5,1),
                new AgedBrie("Aged Brie",5,49),
                new AgedBrie("Aged Brie",5,51),
                new AgedBrie("Aged Brie",7,0),
                new AgedBrie("Aged Brie",7,1),
                new AgedBrie("Aged Brie",7,49),
                new AgedBrie("Aged Brie",7,51),
                new AgedBrie("Aged Brie",11,0),
                new AgedBrie("Aged Brie",11,1),
                new AgedBrie("Aged Brie",11,49),
                new AgedBrie("Aged Brie",11,51),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",-1,0),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",-1,1),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",-1,49),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",-1,51),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",1,0),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",1,1),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",1,49),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",1,51),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",5,0),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",5,1),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",5,49),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",5,51),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",7,0),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",7,1),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",7,49),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",7,51),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",11,0),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",11,1),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",11,49),
                new BackstagePasses("Backstage passes to a TAFKAL80ETC concert",11,51),
                new Default("Elixir of the Mongoose",-1,0),
                new Default("Elixir of the Mongoose",-1,1),
                new Default("Elixir of the Mongoose",-1,49),
                new Default("Elixir of the Mongoose",-1,51),
                new Default("Elixir of the Mongoose",1,0),
                new Default("Elixir of the Mongoose",1,1),
                new Default("Elixir of the Mongoose",1,49),
                new Default("Elixir of the Mongoose",1,51),
                new Default("Elixir of the Mongoose",5,0),
                new Default("Elixir of the Mongoose",5,1),
                new Default("Elixir of the Mongoose",5,49),
                new Default("Elixir of the Mongoose",5,51),
                new Default("Elixir of the Mongoose",7,0),
                new Default("Elixir of the Mongoose",7,1),
                new Default("Elixir of the Mongoose",7,49),
                new Default("Elixir of the Mongoose",7,51),
                new Default("Elixir of the Mongoose",11,0),
                new Default("Elixir of the Mongoose",11,1),
                new Default("Elixir of the Mongoose",11,49),
                new Default("Elixir of the Mongoose",11,51),
                new Sulfuras("Sulfuras, Hand of Ragnaros",-1,0),
                new Sulfuras("Sulfuras, Hand of Ragnaros",-1,1),
                new Sulfuras("Sulfuras, Hand of Ragnaros",-1,49),
                new Sulfuras("Sulfuras, Hand of Ragnaros",-1,51),
                new Sulfuras("Sulfuras, Hand of Ragnaros",1,0),
                new Sulfuras("Sulfuras, Hand of Ragnaros",1,1),
                new Sulfuras("Sulfuras, Hand of Ragnaros",1,49),
                new Sulfuras("Sulfuras, Hand of Ragnaros",1,51),
                new Sulfuras("Sulfuras, Hand of Ragnaros",5,0),
                new Sulfuras("Sulfuras, Hand of Ragnaros",5,1),
                new Sulfuras("Sulfuras, Hand of Ragnaros",5,49),
                new Sulfuras("Sulfuras, Hand of Ragnaros",5,51),
                new Sulfuras("Sulfuras, Hand of Ragnaros",7,0),
                new Sulfuras("Sulfuras, Hand of Ragnaros",7,1),
                new Sulfuras("Sulfuras, Hand of Ragnaros",7,49),
                new Sulfuras("Sulfuras, Hand of Ragnaros",7,51),
                new Sulfuras("Sulfuras, Hand of Ragnaros",11,0),
                new Sulfuras("Sulfuras, Hand of Ragnaros",11,1),
                new Sulfuras("Sulfuras, Hand of Ragnaros",11,49),
                new Sulfuras("Sulfuras, Hand of Ragnaros",11,51),
                new ManaCake("Conjured Mana Cake",-1,0),
                new ManaCake("Conjured Mana Cake",-1,1),
                new ManaCake("Conjured Mana Cake",-1,49),
                new ManaCake("Conjured Mana Cake",-1,51),
                new ManaCake("Conjured Mana Cake",1,0),
                new ManaCake("Conjured Mana Cake",1,1),
                new ManaCake("Conjured Mana Cake",1,49),
                new ManaCake("Conjured Mana Cake",1,51),
                new ManaCake("Conjured Mana Cake",5,0),
                new ManaCake("Conjured Mana Cake",5,1),
                new ManaCake("Conjured Mana Cake",5,49),
                new ManaCake("Conjured Mana Cake",5,51),
                new ManaCake("Conjured Mana Cake",7,0),
                new ManaCake("Conjured Mana Cake",7,1),
                new ManaCake("Conjured Mana Cake",7,49),
                new ManaCake("Conjured Mana Cake",7,51),
                new ManaCake("Conjured Mana Cake",11,0),
                new ManaCake("Conjured Mana Cake",11,1),
                new ManaCake("Conjured Mana Cake",11,49),
                new ManaCake("Conjured Mana Cake",11,51)
            };

            CombinationApprovals.VerifyAllCombinations(UpdateQualityTestSender, testList);
        }

        private string UpdateQualityTestSender(CompleteItem item)
        {
            List<CompleteItem> items = new List<CompleteItem> { item };
            GildedRoseService app = new GildedRoseService();
            app.SetItems(items);

            app.UpdateQuality();

            return app.GetItems()[0].ToString();
        }
    }
}
