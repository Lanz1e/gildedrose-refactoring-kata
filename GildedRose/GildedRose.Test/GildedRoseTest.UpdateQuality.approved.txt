[+5 Dexterity Vest, -2, 0] => +5 Dexterity Vest, -2, 0
[+5 Dexterity Vest, -2, 0] => +5 Dexterity Vest, -2, 0
[+5 Dexterity Vest, -2, 47] => +5 Dexterity Vest, -2, 47
[+5 Dexterity Vest, -2, 49] => +5 Dexterity Vest, -2, 49
[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, 0, 0
[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, 0, 0
[+5 Dexterity Vest, 0, 48] => +5 Dexterity Vest, 0, 48
[+5 Dexterity Vest, 0, 50] => +5 Dexterity Vest, 0, 50
[+5 Dexterity Vest, 4, 0] => +5 Dexterity Vest, 4, 0
[+5 Dexterity Vest, 4, 0] => +5 Dexterity Vest, 4, 0
[+5 Dexterity Vest, 4, 48] => +5 Dexterity Vest, 4, 48
[+5 Dexterity Vest, 4, 50] => +5 Dexterity Vest, 4, 50
[+5 Dexterity Vest, 6, 0] => +5 Dexterity Vest, 6, 0
[+5 Dexterity Vest, 6, 0] => +5 Dexterity Vest, 6, 0
[+5 Dexterity Vest, 6, 48] => +5 Dexterity Vest, 6, 48
[+5 Dexterity Vest, 6, 50] => +5 Dexterity Vest, 6, 50
[+5 Dexterity Vest, 10, 0] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest, 10, 0] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest, 10, 48] => +5 Dexterity Vest, 10, 48
[+5 Dexterity Vest, 10, 50] => +5 Dexterity Vest, 10, 50
[Aged Brie, -2, 2] => Aged Brie, -2, 2
[Aged Brie, -2, 3] => Aged Brie, -2, 3
[Aged Brie, -2, 50] => Aged Brie, -2, 50
[Aged Brie, -2, 51] => Aged Brie, -2, 51
[Aged Brie, 0, 1] => Aged Brie, 0, 1
[Aged Brie, 0, 2] => Aged Brie, 0, 2
[Aged Brie, 0, 50] => Aged Brie, 0, 50
[Aged Brie, 0, 51] => Aged Brie, 0, 51
[Aged Brie, 4, 1] => Aged Brie, 4, 1
[Aged Brie, 4, 2] => Aged Brie, 4, 2
[Aged Brie, 4, 50] => Aged Brie, 4, 50
[Aged Brie, 4, 51] => Aged Brie, 4, 51
[Aged Brie, 6, 1] => Aged Brie, 6, 1
[Aged Brie, 6, 2] => Aged Brie, 6, 2
[Aged Brie, 6, 50] => Aged Brie, 6, 50
[Aged Brie, 6, 51] => Aged Brie, 6, 51
[Aged Brie, 10, 1] => Aged Brie, 10, 1
[Aged Brie, 10, 2] => Aged Brie, 10, 2
[Aged Brie, 10, 50] => Aged Brie, 10, 50
[Aged Brie, 10, 51] => Aged Brie, 10, 51
[Backstage passes to a TAFKAL80ETC concert, -2, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert, -2, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert, -2, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert, -2, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 3] => Backstage passes to a TAFKAL80ETC concert, 0, 3
[Backstage passes to a TAFKAL80ETC concert, 0, 4] => Backstage passes to a TAFKAL80ETC concert, 0, 4
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, 0, 51
[Backstage passes to a TAFKAL80ETC concert, 4, 3] => Backstage passes to a TAFKAL80ETC concert, 4, 3
[Backstage passes to a TAFKAL80ETC concert, 4, 4] => Backstage passes to a TAFKAL80ETC concert, 4, 4
[Backstage passes to a TAFKAL80ETC concert, 4, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50
[Backstage passes to a TAFKAL80ETC concert, 4, 51] => Backstage passes to a TAFKAL80ETC concert, 4, 51
[Backstage passes to a TAFKAL80ETC concert, 6, 2] => Backstage passes to a TAFKAL80ETC concert, 6, 2
[Backstage passes to a TAFKAL80ETC concert, 6, 3] => Backstage passes to a TAFKAL80ETC concert, 6, 3
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 6, 50
[Backstage passes to a TAFKAL80ETC concert, 6, 51] => Backstage passes to a TAFKAL80ETC concert, 6, 51
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => Backstage passes to a TAFKAL80ETC concert, 10, 2
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 10, 51
[Elixir of the Mongoose, -2, 0] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose, -2, 0] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose, -2, 47] => Elixir of the Mongoose, -2, 47
[Elixir of the Mongoose, -2, 49] => Elixir of the Mongoose, -2, 49
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, 0, 0
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, 0, 0
[Elixir of the Mongoose, 0, 48] => Elixir of the Mongoose, 0, 48
[Elixir of the Mongoose, 0, 50] => Elixir of the Mongoose, 0, 50
[Elixir of the Mongoose, 4, 0] => Elixir of the Mongoose, 4, 0
[Elixir of the Mongoose, 4, 0] => Elixir of the Mongoose, 4, 0
[Elixir of the Mongoose, 4, 48] => Elixir of the Mongoose, 4, 48
[Elixir of the Mongoose, 4, 50] => Elixir of the Mongoose, 4, 50
[Elixir of the Mongoose, 6, 0] => Elixir of the Mongoose, 6, 0
[Elixir of the Mongoose, 6, 0] => Elixir of the Mongoose, 6, 0
[Elixir of the Mongoose, 6, 48] => Elixir of the Mongoose, 6, 48
[Elixir of the Mongoose, 6, 50] => Elixir of the Mongoose, 6, 50
[Elixir of the Mongoose, 10, 0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose, 10, 0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose, 10, 48] => Elixir of the Mongoose, 10, 48
[Elixir of the Mongoose, 10, 50] => Elixir of the Mongoose, 10, 50
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0
[Sulfuras, Hand of Ragnaros, 1, 1] => Sulfuras, Hand of Ragnaros, 1, 1
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49
[Sulfuras, Hand of Ragnaros, 1, 51] => Sulfuras, Hand of Ragnaros, 1, 51
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49
[Sulfuras, Hand of Ragnaros, 5, 51] => Sulfuras, Hand of Ragnaros, 5, 51
[Sulfuras, Hand of Ragnaros, 7, 0] => Sulfuras, Hand of Ragnaros, 7, 0
[Sulfuras, Hand of Ragnaros, 7, 1] => Sulfuras, Hand of Ragnaros, 7, 1
[Sulfuras, Hand of Ragnaros, 7, 49] => Sulfuras, Hand of Ragnaros, 7, 49
[Sulfuras, Hand of Ragnaros, 7, 51] => Sulfuras, Hand of Ragnaros, 7, 51
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49
[Sulfuras, Hand of Ragnaros, 11, 51] => Sulfuras, Hand of Ragnaros, 11, 51
[Conjured Mana Cake, -2, 0] => Conjured Mana Cake, -2, 0
[Conjured Mana Cake, -2, 0] => Conjured Mana Cake, -2, 0
[Conjured Mana Cake, -2, 45] => Conjured Mana Cake, -2, 45
[Conjured Mana Cake, -2, 47] => Conjured Mana Cake, -2, 47
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, 0, 0
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, 0, 0
[Conjured Mana Cake, 0, 47] => Conjured Mana Cake, 0, 47
[Conjured Mana Cake, 0, 49] => Conjured Mana Cake, 0, 49
[Conjured Mana Cake, 4, 0] => Conjured Mana Cake, 4, 0
[Conjured Mana Cake, 4, 0] => Conjured Mana Cake, 4, 0
[Conjured Mana Cake, 4, 47] => Conjured Mana Cake, 4, 47
[Conjured Mana Cake, 4, 49] => Conjured Mana Cake, 4, 49
[Conjured Mana Cake, 6, 0] => Conjured Mana Cake, 6, 0
[Conjured Mana Cake, 6, 0] => Conjured Mana Cake, 6, 0
[Conjured Mana Cake, 6, 47] => Conjured Mana Cake, 6, 47
[Conjured Mana Cake, 6, 49] => Conjured Mana Cake, 6, 49
[Conjured Mana Cake, 10, 0] => Conjured Mana Cake, 10, 0
[Conjured Mana Cake, 10, 0] => Conjured Mana Cake, 10, 0
[Conjured Mana Cake, 10, 47] => Conjured Mana Cake, 10, 47
[Conjured Mana Cake, 10, 49] => Conjured Mana Cake, 10, 49
